Patch MagpieRSS for basic use in PHP 8

opengeek opened the pull request:
### What does it do?
Patch MagpieRSS to work in PHP 8

### Why is it needed?
News and Security feeds are currently blank in PHP 8.

### How to test
Load the Manager Dashboard in a PHP 8 environment and confirm the News and Security feeds are not blank.

### Related issue(s)/PR(s)
Resolves #15712 

### Note
Though this makes it work for now in PHP 8, it is not an ideal solution because it bypasses MagpieRSS cache features to get around the fact that in PHP 8 you cannot serialize an XMLParser instance.

opengeek committed:
Patch MagpieRSS for basic use in PHP 8

Mark-H committed:
Add 1-hour cache to the dashboard feeds

Mark-H commented:
Tested and confirmed on PHP8, tho I did [send a separate PR](https://github.com/opengeek/revolution/pull/29) to add back some caching behavior for consideration.

opengeek committed:
Add 1-hour cache to the dashboard feeds

Add 1-hour cache to the dashboard feeds

Merge pull request #29 from Mark-H/patch-magpie
